### ISSUE TITLE
Add a What-If/Verify page

### ISSUE THREAD

**Issue by kungfu71186** [firefly-iii/firefly-iii#3512]
**Description**
Currently, when you import a csv file, there is no way to see what will actually happen and verify before the changes are submitted. It would be nice to have a what-if scenario or a verify to see what is actually going to happen. This includes showing all fields specified when mapping the columns. 

**Solution**
Have an option to verify the transactions before doing a final submit. Once you verify, you can then click a button to actually commit the transactions. 

Basically:
Start > Configure > Roles > Run + Verify > Run + Commit

Replace the initial run page with a verify page (the page where you can download the config file and it says start job)

**Comment by JC5** [comment]
No, sorry. There are a lot of edge cases where the ultimate choice is made by Firefly III. Plus, account names, transaction types other details are also influenced by your transaction rules. Which means that ultimately the CSV importer won’t really be able to tell you what the transaction will look like. That removes the value of such an extra step.

**Comment by kungfu71186** [comment]
Would it be possible to create a DB transaction and verify and then commit? Also only show the required fields if that's the case? Like date, amount, description and then the type of transaction like it currently does?

**Comment by JC5** [comment]
That's not how the CSV importer works, I'm afraid. It picks up your transaction and puts everything into a JSON array. This is pretty much copy and paste. Mapped values are replaced by ID's. That's sent to Firefly III for further processing.

So no, sorry, it wouldn't be possible.

**Comment by kungfu71186** [comment]
Ahhh I see. Thanks for looking into it though

**Comment by kungfu71186** [comment]
Just to expand on this a bit. It would be nice to see the required fields. I think everything is there except for date. So, something like:

https://github.com/firefly-iii/csv-importer/blob/5ed6e12e3873032b1a49150165fae1e344443af4/app/Services/Import/Routine/APISubmitter.php#L260
```
                    $message = sprintf(
                        'Created %s <a target="_blank" href="%s">#%d %s "%s"</a> (%s %s)',
                        $transaction->type,
                        sprintf('%s/transactions/show/%d', env('FIREFLY_III_URI'), $group->id),
                        $group->id,
                        $transaction->date,
                        e($transaction->description),
                        $transaction->currencyCode,
                        round($transaction->amount, $transaction->currencyDecimalPlaces)
                    );
```

Not sure about formatting though, but probably use the same formatting that was specified in the configuration from the import or whatever is returned from the response if it's a string already.
